test/extended/prometheus/prometheus.go

```diff
@@ -105,6 +107,108 @@ var _ = g.Describe("[sig-instrumentation][Late] OpenShift alerting rules [apigro
 		"ClusterVersionOperatorDown",
 	)
 
+	// These alerts are known to have metrics that may be absent or metric
+	// selectors returning empty results. In most cases, these are errors or
+	// conditional metrics which are only present if an error or a certain
+	// condition occurs. In these cases, we can skip the validation of the
+	// alert since we won't be able to easily tell the difference between what
+	// is expected and what isn't.
+	transientAlertExceptions := sets.NewString(
```

Comment:
Given that we have so many exceptions, I'm not convinced by the usefulness of the test. I fear that we'll be constantly bugged to add new alerting rules to this list...

Reply:
ACK, should I close this PR (and only validate the collection-based metrics on CMO side)?

Reply:
I think so :bow: